libtransact/src/contract/address/error.rs

```diff
@@ -13,24 +13,14 @@
 // limitations under the License.
 
 #[derive(Debug)]
-pub enum AddresserError {
-    KeyHashAddresserError(String),
-    DoubleKeyHashAddresserError(String),
-    TripleKeyHashAddresserError(String),
+pub struct AddresserError {
+    pub message: String,
 }
 
 impl std::fmt::Display for AddresserError {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        match *self {
-            AddresserError::KeyHashAddresserError(ref s) => {
-                write!(f, "Unable to compute hash: {}", s.to_string())
-            }
-            AddresserError::DoubleKeyHashAddresserError(ref s) => {
-                write!(f, "Unable to compute hash: {}", s.to_string())
-            }
-            AddresserError::TripleKeyHashAddresserError(ref s) => {
-                write!(f, "Unable to compute hash: {}", s.to_string())
-            }
+        match self {
+            AddresserError { message } => write!(f, "Invalid hash length: {}", message),
```

Comment:
Does "Invalid hash length" always make sense in this context?

Reply:
Not any more, addressed.